### PR TITLE
doc: add release document

### PR DESCRIPTION
**Pull request by hakehuang**
put release report documnet here so that more audiences can share
their comments.

please create issue in test_result repo for any comments.

Signed-off-by: Hake Huang <hake.huang@oss.nxp.com>

### CODE REVIEW

**Comment by hakehuang** [comment]
@nashif  would this be ok?

**Review by nashif — CHANGES_REQUESTED** [review]
I would rather see the documents in text format( for example .rst) and not microsoft doc.

**Comment by hakehuang** [comment]
> I would rather see the documents in text format( for example .rst) and not microsoft doc.

@nashif  update to rcs format, please check.

File: doc/reports/prelude.rst [hunk 1]
```diff
@@ -0,0 +1,82 @@
+
+Zephyr Test Result Report
+############################
+
+
+OVERVIEW
```
Comment (nashif) [inline doc/reports/prelude.rst:6]: do not use all caps

File: doc/reports/prelude.rst [hunk 2]
```diff
@@ -0,0 +1,82 @@
+
+Zephyr Test Result Report
+############################
+
+
+OVERVIEW
+************************************
```
Comment (nashif) [inline doc/reports/prelude.rst:7]: this should be same length as the title.

File: doc/reports/summary.rst [hunk 3]
```diff
@@ -0,0 +1,208 @@
+
+TEST RESULTS
```
Comment (nashif) [inline doc/reports/summary.rst:2]: do not use all caps

**Review by nashif — CHANGES_REQUESTED** [review]
split into multiple commits, one adding the doc infra and another adding the actual doc.

File: doc/reports/prelude.rst (hunk 1 above)
Comment (hakehuang) [reply doc/reports/prelude.rst:6]: ok, done

File: doc/reports/prelude.rst (hunk 2 above)
Comment (hakehuang) [reply doc/reports/prelude.rst:7]: ok, done.

File: doc/reports/summary.rst (hunk 3 above)
Comment (hakehuang) [reply doc/reports/summary.rst:2]: ok, done

**Review by nashif — APPROVED** [review]

**Comment by abrodkin** [comment]
Looks like LaTeX is being used for building, while from my experience it's always a pain to get LaTeX distribution installed.
Thus I would suggest to use `rst2pdf` (https://rst2pdf.org/). Its installation is as simple as `pip install rst2pdf`. Though some quite trivial changes needs to be done in `doc/conf.py`:
```python
extensions = [
    # ...
    'rst2pdf.pdfbuilder',
]

# Grouping the document tree into PDF files. List of tuples
# (source start file, target name, title, author, options).
pdf_documents = [
    ('index', 'MyProject', 'My Project', 'Author Name'),
]
```
Or see more details in the official manual: https://rst2pdf.org/static/manual.pdf.

**Comment by hakehuang** [comment]
> Looks like LaTeX is being used for building, while from my experience it's always a pain to get LaTeX distribution installed.
> Thus I would suggest to use `rst2pdf` (https://rst2pdf.org/). Its installation is as simple as `pip install rst2pdf`. Though some quite trivial changes needs to be done in `doc/conf.py`:
> 
> ```python
> extensions = [
>     # ...
>     'rst2pdf.pdfbuilder',
> ]
> 
> # Grouping the document tree into PDF files. List of tuples
> # (source start file, target name, title, author, options).
> pdf_documents = [
>     ('index', 'MyProject', 'My Project', 'Author Name'),
> ]
> ```
> 
> Or see more details in the official manual: https://rst2pdf.org/static/manual.pdf.

the tools installation is based on below guider. I am not generating pdf but html format, though. let me try you method
https://www.sphinx-doc.org/en/master/usage/installation.html

**Comment by hakehuang** [comment]
@abrodkin I add rst2pdf support, please check

**Comment by abrodkin** [comment]
Thanks, @hakehuang!

FWIW that's what I got generated with:
```
$ make pdf

Running Sphinx v2.0.0+/5a1e3ff
making output directory... done
loading pickled environment... done
building [mo]: targets for 0 po files that are out of date
building [pdf]: targets for 3 source files that are out of date
updating environment: 0 added, 0 changed, 0 removed
looking for now-outdated files... none found
processing ZephyrQualityManagement...
index
reports/prelude
reports/summary
resolving references...
done
writing ZephyrQualityManagement...
done
build succeeded.
```
[ZephyrQualityManagement.pdf](https://github.com/zephyrproject-rtos/test_results/files/5843881/ZephyrQualityManagement.pdf)

File: doc/reports/prelude.rst [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+Intended audience
+=================
+
+The report can be used by anyone interested in the health of Zephyr, like TSC members, Zephyr community developers, and stakeholders considering equipping their platforms with the Zephyr RTOS. 
+
+
+Legal
+=====
+
+Testing WG community does not make any representations or warranties as to the accuracy of the information, quality, reliability, functionality or compatibility. Names and brands in that report may be claimed as the property of others. Our Testing WG community members don't receive any commercial benefit from the board’s manufacturers. We obtain test results using boards which we have on hands during our daily work. If you want your platform results to be added here, you can send your test results by contacting the Testing WG community using that email Testing community will not verify data provided by you and doesn’t make any claims about the reliability and correctness of it. We understand that performance and all test results vary depending on system configuration. Provided data below can’not be used in any marketing activities or advertisements. No certification issued for all results which you can view in that document.
```
Comment (LeiW000) [inline doc/reports/prelude.rst:59]: "Provided data below can'not be....."

" ' " should not removed.

File: doc/reports/summary.rst [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+|New FW/MW           | 1             | 1                     | 0         | 100       |
++--------------------+---------------+-----------------------+-----------+-----------+
+|update FW/MW        | 2             | 2                     | 0         | 100       |
++--------------------+---------------+-----------------------+-----------+-----------+
+|Tools               | 3             | 3                     | 0         | 100       |
++--------------------+---------------+-----------------------+-----------+-----------+
+|Document            | 1             | 1                     | 0         | 100       |
++--------------------+---------------+-----------------------+-----------+-----------+
+
+.. _API revision: https://docs.zephyrproject.org/latest/releases/release-notes-2.4.html#api-changes
```
Comment (LeiW000) [inline doc/reports/summary.rst:55]: Should be "2.5"?

**Review by LeiW000 — CHANGES_REQUESTED** [review]

File: doc/reports/prelude.rst [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+
+Overview
+********
+
+This report provides the most up to date information about the health and quality of the current Zephyr RTOS version  . This report gathers key metrics describing the state and performance of the current version and the results of its testing. Whenever applied, references for detailed analysis are provided. This report is a part of the Quality Assurance process for the Zephyr RTOS product.
+
+
+Word format file can be found `here
+<https://docs.google.com/document/d/12kEprn7uBkD238EZmQkvD9_hHV9ttcea8Ke-MszLRnc/edit>`_
```
Comment (abrodkin) [inline doc/reports/prelude.rst:13]: Document by that URL is long outdated. And why mention it here at all if this text version is supposed to be used by people outside of Testing WG?

File: doc/reports/prelude.rst [hunk 7]
```diff
@@ -0,0 +1,82 @@
+
+Zephyr Test Result Report
+############################
+
+
+Overview
+********
+
+This report provides the most up to date information about the health and quality of the current Zephyr RTOS version  . This report gathers key metrics describing the state and performance of the current version and the results of its testing. Whenever applied, references for detailed analysis are provided. This report is a part of the Quality Assurance process for the Zephyr RTOS product.
```
Comment (abrodkin) [inline doc/reports/prelude.rst:9]: Extra spaces:
> ... Zephyr RTOS version __.

**Review by abrodkin — CHANGES_REQUESTED** [review]

File: doc/reports/summary.rst (hunk 5 above)
Comment (hakehuang) [reply doc/reports/summary.rst:55]: we have no 2.5 data yet

File: doc/reports/prelude.rst (hunk 4 above)
Comment (hakehuang) [reply doc/reports/prelude.rst:59]: ok

File: doc/reports/prelude.rst (hunk 6 above)
Comment (hakehuang) [reply doc/reports/prelude.rst:13]: ok, I remove it

File: doc/reports/prelude.rst (hunk 7 above)
Comment (hakehuang) [reply doc/reports/prelude.rst:9]: fix it

**Pull request merged**